### ISSUE TITLE
az keyvault purge gives error "The resource type 'deletedVaults' could not be found in the namespace Microsoft.KeyVault"

### ISSUE THREAD

**Issue by holtalanm** [Azure/azure-cli#18126]
**Describe the bug**
In previous versions of azure cli, running `az keyvault purge --name <vault name>` for a deleted key vault would successfully delete/purge the soft-deleted key vault.

In this version (2.23.0), running the above command gives the following error:
```
(InvalidResourceType) The resource type 'deletedVaults' could not be found in the namespace 'Microsoft.KeyVault' 
for api version '2021-04-01-preview'. The supported api-versions are 
'2016-10-01,2018-02-14-preview,2018-02-14,2019-09-01'.
```

**To Reproduce**
Delete a resource group that contains a key vault.
run `az keyvault purge --name <vault name>` for that key vault to purge it.

**Expected behavior**
vault is purged successfully

**Environment summary**
az --version
azure-cli                         2.23.0

core                              2.23.0
telemetry                          1.0.6

Python location 'C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\python.exe'
Extensions directory 'C:\Users\AHolt\.azure\cliextensions'

Python (Windows) 3.8.9 (tags/v3.8.9:a743f81, Apr  6 2021, 13:22:56) [MSC v.1928 32 bit (Intel)]

Legal docs and information: aka.ms/AzureCliLegal

Your CLI is up-to-date.

Please let us know how we are doing: https://aka.ms/azureclihats
and let us know if you're interested in trying out our newest features: https://aka.ms/CLIUXstudy

**Additional Information**

I am running this against AzureUSGov cloud.

**Comment by yonzhan** [comment]
keyvault

**Comment by evelyn-ys** [comment]
Hi @holtalanm, Can you check with the latest CLI if you can reproduce it?

**Comment by yonzhan** [comment]
Hi @holtalanm, any update?

**Comment by holtalanm** [comment]
give me a minute.  would need to install latest cli to test.

**Comment by holtalanm** [comment]
looks like it is fixed on 2.25.0

**Comment by holtalanm** [comment]
I take it back.  Just got this error this morning on 2.25.0:

```
(InvalidResourceType) The resource type 'deletedVaults' could not be found in the namespace 'Microsoft.KeyVault' for api version '2021-04-01-preview'. The supported api-versions are '2016-10-01,2018-02-14-preview,2018-02-14,2019-09-01'.
```

**Comment by evelyn-ys** [comment]
@holtalanm May I know which cloud are you in and where does the deleted vault located in?
The error msg is from service side so I guess you need to check keyvault's doc for support details for different api version in each cloud/location...

**Comment by holtalanm** [comment]
cloud:  Azure US Gov
region: Virginia

**Comment by holtalanm** [comment]
the cli was originally installed following this guide:  https://docs.microsoft.com/en-us/azure/azure-government/documentation-government-get-started-connect-with-cli#install-azure-cli

no mention of api compatibility anywhere.

**Comment by evelyn-ys** [comment]
The situation is the error msg is returned by service side. It's OK to install CLI with the guide above since it's a client tool.
Seems like your scenario similar to #18006. One week ago service side is working on 2021-04-01-preview support on Fairfax. Could you test again with latest CLI? If still not work, I'll contact service team to take a look.

Sorry for the inconvenience😣

**Comment by holtalanm** [comment]
Every time I test this, I have to update, then when it is broken, I have to re-install my old version again.  I honestly don't have time to do that over and over again.  I already installed the updated version once and saw it was broken, then rolled back and submitted this Issue on GitHub.  Then I installed it again when asked (after it sat here for over half a month) and confirmed it was still broken, then rolled back.  

If I'm understanding, now I'm being asked _a third time_ to install the latest version of the cli _just to confirm something I already confirmed_?

I understand the message is service-side, but clearly there is something wonky with the client side since reinstalling the older version of the CLI caused everything to work again.


TL;DR.  I'm not going to keep reinstalling a broken version of the CLI just to keep saying "yes, it is still broken".

**Comment by evelyn-ys** [comment]
@holtalanm So so sorry about the bad experience. Your feelings are totally received and considered.

The newer version of CLI has updated the service API version which service asked. There does exist latency for service ready in USGov. We kept asking your retry because service might be ready now and newer CLI version with new service api version will work as normal.

@amitbapat Could you check with service team If it's ready in USGov for 'deletedVaults' with api version '2021-04-01-preview'?
Here's the previous error returned from service side:
```
(InvalidResourceType) The resource type 'deletedVaults' could not be found in the namespace 'Microsoft.KeyVault' 
for api version '2021-04-01-preview'. The supported api-versions are 
'2016-10-01,2018-02-14-preview,2018-02-14,2019-09-01'.
```
